src/main/java/com/gtnewhorizons/navigator/api/model/steps/UniversalInteractableStep.java

```diff
@@ -0,0 +1,57 @@
+package com.gtnewhorizons.navigator.api.model.steps;
+
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.GuiScreen;
+
+import com.gtnewhorizons.navigator.api.journeymap.drawsteps.JMInteractableStep;
+import com.gtnewhorizons.navigator.api.model.locations.IWaypointAndLocationProvider;
+import com.gtnewhorizons.navigator.api.xaero.rendersteps.XaeroInteractableStep;
+
+import cpw.mods.fml.common.Optional;
+
+@Optional.Interface(
+    iface = "com.gtnewhorizons.navigator.api.journeymap.drawsteps.JMInteractableStep",
+    modid = "journeymap")
```

Comment:
Should `@Optional` annotation be added for type for Xaero's Map, shouldn't it?

Reply:
The problem is that the JMRenderSteps extend the Journeymap DrawStep while the Xaero ones don't extend anything outside of this mod.

Reply:
Perhaps such an optional interface should be added to Xaero's map-related classes to allow such access to map mod internals *in the future*. But I leave this design choice up to you.

Reply:
As someone that have spent a ton of time dealing with Xaero code, I can say with 100% certainty that it is definitely not desired by anyone. We are way better off designing things inhouse and forcing Xaero to work with it.